pkg/controller/elasticsearch/filesettings/reconciler.go

```diff
@@ -60,18 +61,18 @@ func ReconcileSecret(
 	ctx context.Context,
 	c k8s.Client,
 	expected corev1.Secret,
-	es esv1.Elasticsearch,
+	owner client.Object,
 ) error {
 	reconciled := &corev1.Secret{}
 	return reconciler.ReconcileResource(reconciler.Params{
 		Context:    ctx,
 		Client:     c,
-		Owner:      &es,
+		Owner:      owner,
 		Expected:   &expected,
 		Reconciled: reconciled,
 		NeedsUpdate: func() bool {
-			return !maps.IsSubset(expected.Labels, reconciled.Labels) ||
-				!maps.IsSubset(expected.Annotations, reconciled.Annotations) ||
+			return !maps.IsEqualSubset(expected.Labels, reconciled.Labels, managedLabels) ||
```

Comment:
If `IsEqualSubset` returns `true` when *not* equal do we need the `! `?

I can observe the following messages every time I start the operator while the `Secret` `sample-es-file-settings` is not updated:

```
2023-12-13T08:42:46.849+0100    INFO    stackconfigpolicy-controller    Updating resource       {"service.version": "0.0.0-SNAPSHOT+00000000", "iteration": "1", "namespace": "elastic-system", "policy_name": "test-stack-config-policy", "kind": "Secret", "namespace": "test-ns", "name": "sample-es-file-settings"}
```

*restart*

```
2023-12-13T08:44:52.461+0100    INFO    stackconfigpolicy-controller    Updating resource       {"service.version": "0.0.0-SNAPSHOT+00000000", "iteration": "1", "namespace": "elastic-system", "policy_name": "test-stack-config-policy", "kind": "Secret", "namespace": "test-ns", "name": "sample-es-file-settings"}
```

Reply:
sorry 🤦  fixing this.

Reply:
@barkbay @pebrc please take another look, I changed the condition a bit so this keeps existing behaviour and also checks if managed fields have changed/removed.

Reply:
I don't see unnecessary updates being sent in my testing now